lib/valkey.rb

```diff
@@ -182,15 +182,32 @@ def convert_response(res, &block)
   end
 
   def send_command(command_type, command_args = [], &block)
+    # Validate connection
+    if @connection.nil?
+      raise "Connection is nil"
+    elsif @connection.null?
+      raise "Connection pointer is null"
+    elsif @connection.address.zero?
+      raise "Connection address is 0"
+    end
```

Comment:
This block runs on eacy command, so when/how those methods can turn to invalid after creating the connection?